### PR TITLE
Setup data refresh for Dashboard

### PR DESCRIPTION
**Pull request by nitish1295**
- Added a selector component RefreshDataComponent
- Added changes to Dashboard to re-fetch data
- Update to create_fake_data.py to generate data with a delay

### CODE REVIEW

File: webapp/src/component/RefreshDataComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+ * @param {event} SelectChangeEvent - Change event
+ * @returns {Promise<Object>} - The data fetched from the endpoint.
+ *
+ * @see fetchData
+ */
+
+export default function RefreshDataComponent() {
+  const theme = useTheme();
+  const colors = tokens(theme.palette.mode);
+  const { refreshInterval, setrRefreshInterval } = useConfigurationContext();
```
Comment (Dref360) [inline webapp/src/component/RefreshDataComponent.tsx]: Small typo `setrRefreshInterval => setRefreshInterval`

File: webapp/src/component/RefreshDataComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      <Button onClick={handleOpen} color="secondary">
+        Select Refresh Time
+      </Button>
+      <Modal
+        open={open}
+        onClose={handleClose}
+        aria-labelledby="modal-modal-title"
+        aria-describedby="modal-modal-description"
+      >
+        <Box sx={style}>
```
Comment (Dref360) [inline webapp/src/component/RefreshDataComponent.tsx:75]: [Misc] To be more expressive, you can use `styled` here.

```
const IntervalSelectBox = styled(Box)({
    position: "absolute" as "absolute",
    top: "50%",
    left: "50%",
    transform: "translate(-50%, -50%)",
    width: 400,
    bgcolor: colors.primary["400"],
    border: "2px solid #000",
    boxShadow: 24,
    p: 4,
  });

...

	<IntervalSelectBox>
	...
	</IntervalSelectBox>
```

File: webapp/src/context/Configuration.tsx [hunk 3]
```diff
@@ -3,6 +3,8 @@ import React, { createContext, useContext, useState } from "react";
 interface ConfigurationContextValue {
   runId: string | null;
   setRunId: React.Dispatch<React.SetStateAction<string | null>>;
+  refreshInterval: string | undefined;
```
Comment (Dref360) [inline webapp/src/context/Configuration.tsx]: Should this be a Number instead?

**Review by Dref360 — CHANGES_REQUESTED** [review]
Very nice! 

Small comments, feel free to ignore the [Misc] one.

File: webapp/src/component/RefreshDataComponent.tsx (hunk 1 above)
Comment (nitish1295) [reply webapp/src/component/RefreshDataComponent.tsx]: Fixed

File: webapp/src/context/Configuration.tsx (hunk 3 above)
Comment (nitish1295) [reply webapp/src/context/Configuration.tsx]: Fixed

File: webapp/src/component/RefreshDataComponent.tsx (hunk 2 above)
Comment (nitish1295) [reply webapp/src/component/RefreshDataComponent.tsx:75]: Will try to look into this next time

**Review by Dref360 — APPROVED** [review]
LGTM

**Pull request merged**